core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -126,11 +127,17 @@ trait Logging {
 
   private def initializeLogging(isInterpreter: Boolean, silent: Boolean): Unit = {
     if (Logging.isLog4j2()) {
-      // If Log4j is used but is not initialized, load a default properties file
-      val log4j2Initialized = !LogManager.getRootLogger
-        .asInstanceOf[org.apache.logging.log4j.core.Logger].getAppenders.isEmpty
+      val rootLogger = LogManager.getRootLogger.asInstanceOf[org.apache.logging.log4j.core.Logger]
+      // If Log4j 2 is used but is initialized by default configuration,
+      // load a default properties file
+      // (see org.apache.logging.log4j.core.config.DefaultConfiguration)
+      val needToInitializeLog4j2 = rootLogger.getAppenders.isEmpty ||
+        (rootLogger.getAppenders.size() == 1 &&
+          rootLogger.getLevel == Level.ERROR &&
```

Comment:
The default level could be configured and maybe not `ERROR`. But I guess once user specifies it, it may meaning they want to set it up so we don't need to load Spark default properties.